cache/refs.go

```diff
@@ -12,24 +12,25 @@ import (
 	"golang.org/x/net/context"
 )
 
-type ImmutableRef interface {
-	Mountable
+// Ref is a reference to cacheable objects.
+type Ref interface {
```

Comment:
@tonistiigi 

Is this common `Ref` interface SGTY?

My motivation for this is readability

Reply:
Yes, but this would not be exposed to solver, right? Because otherwise, solver would need to recast because it needs the missing methods.